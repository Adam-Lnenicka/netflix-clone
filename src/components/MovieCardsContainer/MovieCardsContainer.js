import React from "react";
import { useApiMoviesArraySelector } from "../../store/selectors";
import MovieCard from "../MovieCard/MovieCard";

const MovieCardsContainer = () => {
  const apiMoviesArray = useApiMoviesArraySelector();

  return (
    <div className="card-layout">
      {apiMoviesArray.map((movie) =>
        movie
          .map((m) => (
            <MovieCard
              key={m.id}
              id={m.id}
              title={m.title}
              poster_path={m.poster_path}
              genres={m.genres.map((g) => (
                <span key={g}>{g} /</span>
              ))}
              release_date={m.release_date}
              vote_average={m.vote_average}
              overview={m.overview}
              movie={movie}
              runtime={m.runtime}
              broken={(e) =>
                (e.target.src =
                  "https://images.unsplash.com/photo-1560109947-543149eceb16?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1035&q=80")
              }
            />
          ))
          .slice(0, 10)
      )}
    </div>
  );
};

export default MovieCardsContainer;
