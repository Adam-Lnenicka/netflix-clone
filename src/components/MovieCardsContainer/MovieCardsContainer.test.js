import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import MovieCardsContainer from "./MovieCardsContainer";

describe("MovieCardsContainer", () => {
  const initialState = {};
  const mockStore = configureStore();
  it("MovieCardsContainer matches snapshot", () => {
    const store = mockStore(initialState);
    const props = {
      title: "Captain Marvel",
      genres: ["Animation"],
      release_date: "2016",
      description: "Movie description",
      poster_path: "https://",
      overview: "test description test",
      vote_average: "vote average test",
      movie: "movie test",
      reuntime: "runtime test",
      broken:
        "https://images.unsplash.com/photo-1560109947-543149eceb16?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1035&q=80",
    };
    const wrapper = shallow(
      <Provider store={store}>
        <MovieCardsContainer {...props} />{" "}
      </Provider>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
