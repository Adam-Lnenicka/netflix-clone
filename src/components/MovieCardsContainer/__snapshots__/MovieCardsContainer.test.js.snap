// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardsContainer MovieCardsContainer matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <MovieCardsContainer
    broken="https://images.unsplash.com/photo-1560109947-543149eceb16?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1035&q=80"
    description="Movie description"
    genres={
      Array [
        "Animation",
      ]
    }
    movie="movie test"
    overview="test description test"
    poster_path="https://"
    release_date="2016"
    reuntime="runtime test"
    title="Captain Marvel"
    vote_average="vote average test"
  />
   
</ContextProvider>
`;
